server/pulp/server/managers/content/orphan.py

```diff
@@ -256,13 +259,42 @@ def delete_orphan_content_units_by_type(type_id):
             for non_orphan_id in non_orphan:
                 unit_dict.pop(non_orphan_id)
 
-            # Remove the unit and any references on disk
+            # If there's a list of unit ids to delete, filter the orphan list
+            if content_unit_ids is not None:
+                for orphaned_unit_id in unit_dict.keys():
+                    if orphaned_unit_id not in content_unit_ids:
+                        unit_dict.pop(orphaned_unit_id)
+
+            # Remove the unit, lazy catalog entries, and any content in storage.
             for unit_to_delete in unit_dict.itervalues():
+                OrphanManager.delete_catalog_entry(unit_to_delete.id, type_id)
                 unit_to_delete.delete()
                 if unit_to_delete._storage_path:
                     OrphanManager.delete_orphaned_file(unit_to_delete._storage_path)
 
     @staticmethod
+    def delete_catalog_entry(unit_id, unit_type_id):
+        """
+        Deletes the lazy catalog entries associated with the given list of units.
+
+        :param unit_id:      The content unit id.
+        :type  unit_id:      str
+        :param unit_type_id: The content type of the content unit.
+        :type  unit_type_id: str
+        """
+        catalog_entries = model.LazyCatalogEntry.objects(
+            unit_id=str(unit_id),
+            unit_type_id=str(unit_type_id)
+        )
+        msg = _('Deleting {n} catalog entries for orphaned content unit {type}:{id}')
+        _logger.error(msg.format(
+            n=len(catalog_entries),
```

Comment:
Calling ``len()`` causes the ``QuerySet`` to be evaluated, so results are loaded from the DB into python. This is not worth the expense just for a log message. I wouldn't even call ``.count()`` on it. I suggest leaving out the number, and then you only have to do one database action (the delete) instead of two.